storage/libradb/src/transaction_store/mod.rs

```diff
@@ -68,6 +69,31 @@ impl TransactionStore {
         })
     }
 
+    /// Returns the block metadata carried on the block metadata transaction at or preceding
+    /// `version`, together with the version of the block metadata transaction.
+    /// Returns None if there's no such transaction at or preceding `version` (it's likely the genesis
+    /// version 0).
+    pub fn get_block_metadata(&self, version: Version) -> Result<Option<(Version, BlockMetadata)>> {
+        // Maximum TPS from benchmark is around 1000.
+        const MAX_VERSIONS_TO_SEARCH: usize = 1000 * 3;
+
+        // Linear search via `DB::rev_iter()` here, NOT expecting performance hit, due to the fact
+        // that the iterator caches data block and that there are limited number of transactions in
+        // each block.
+        let mut iter = self.db.rev_iter::<TransactionSchema>(Default::default())?;
+        iter.seek(&version)?;
+        for res in iter.take(MAX_VERSIONS_TO_SEARCH) {
```

Comment:
I was hoping you would use some magic here like 
```
.position(|x| return let Transaction:BlockMetatdata(t) = x).map(|x|..)
```

but TBH I don't know exactly how to write infunctional style